spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/SftpSourceConfiguration.java

```diff
@@ -146,4 +183,33 @@ public SftpRemoteFileTemplate sftpTemplate(SessionFactory<LsEntry> sftpSessionFa
 		return new SftpRemoteFileTemplate(sftpSessionFactory);
 	}
 
+	@ConditionalOnProperty(name = "sftp.listOnly")
+	@IdempotentReceiver("idempotentReceiverInterceptor")
+	@Transformer(inputChannel = "sftpFileListChannel", outputChannel = Source.OUTPUT)
```

Comment:
You know, in the `tasklunchrequest-transformer` project we decided to stay away of the `@Transformer` and even `@ServiceActivator` annotations. Try to use `@StreamListener` and `@SendTo` combination instead.

More info is in the https://github.com/spring-cloud-stream-app-starters/tasklaunchrequest-transform/pull/8

Reply:
using `@StreamListener` and `@SendTo` results in the `IdempotentReceiverInterceptor` not being applied via `@IdempotentReceiver`

Reply:
Good catch, indeed. How does it work with the `@ServiceActivator` though?
Thanks. If it is still with no use, then let's stick with the `@Transformer`!

Reply:
Tests all appear to pass when using `@ServiceActivator` so it seems that would be fine as well. Is there any specific benefit to using `@ServiceActivator` over `@Transformer` here? I used `@Transformer` here since the implementation of this method seemed to match the description of `@Transformer` better